### ISSUE TITLE
Conflict between Flags and Before

### ISSUE THREAD

**Issue by albertoleal** [codegangsta/cli#158]
Hi! I've been using cli and I enjoying it a lot, thank you!

I have one flag (--name) and I have one Before function for the same action. It seems that the I cannot access flags values after Before. 

```go
	{
			Name:        "team-remove",
			Usage:       "team-remove --name <name>",
			Description: "Remove an existing team.",
			Flags: []cli.Flag{
				cli.StringFlag{
					Name:  "name, n",
					Value: "",
					Usage: "Name of the team",
				},
			},
			Before: func(c *cli.Context) error {
				context := &Context{Stdout: os.Stdout, Stdin: os.Stdin}
				if Confirm(context, "Are you sure you want to remove this team?") != true {
					return ErrCommandCancelled
				}
				return nil
			},
			Action: func(c *cli.Context) {
				team := &Team{
					Name:   c.String("name"),
					client: NewClient(&http.Client{}),
				}
				result := team.remove()
				fmt.Println(result)
			},
		},

```

**Comment by jszwedko** [comment]
Indeed, this is a bug that should be addressed by #170

**Comment by jszwedko** [comment]
Closed by #170
